### PR TITLE
digest function based on webcrypto API

### PR DESCRIPTION
**Pull request by keroxp**
- Implemented digest function based on [webcrypto API](https://developer.mozilla.org/en-US/docs/Web/API/Web_Crypto_API)
- I don't know whether this is the best solution for providing digest or cryptographic features in deno
  - There are another way like exposing pure hashing function on "deno" namespace.
- Based on the [discussions](https://github.com/denoland/deno_std/issues/48), I think cryptographic features should be provided by deno core, anyway.
- I understand that more discussions are needed about this problem, because I don't know whether all features that required by webcrypto API can be implemented by Rust
- I believe that crypto api is important for many other core features, TLS, Hashing.
- I'm Rust beginner, thanks!
[Screenshot from Gyazo](https://gyazo.com/1a707867183efeab65ab4fd8b3a46f18)

### CODE REVIEW

**Comment by hayd** [comment]
The ideal thing to do is to build in webcrypto (as mentioned in #48), but I think this is a good stop-gap that will give a forwards-compatible api.

This would already be useful, e.g. for https://github.com/denoland/deno_std/pull/84 (websockets).

File: src/msg.fbs [hunk 1]
```diff
@@ -451,4 +453,13 @@ table RunStatusRes {
   exit_signal: int;
 }
 
+table WebCryptoDigest {
+  algorithm: string;
+  data: [ubyte];
```
[the remark this answers is not in the document]
Comment (keroxp) [reply src/msg.fbs]: @ry SHA1 function is now part of deon_std. So may we not need native digest functions immediately? I probably can make a patch around your first pointing, but don't fully understand what you say at the second. If it is quite hard to merge, I'm ok to drop this PR. How do you think?

**Comment by keroxp** [comment]
This PR can't be merged into latest master. drop it.

**Comment by hayd** [comment]
I think this will be a good approach to add in the future (preferable to algos in ts) using new ops.

File: src/msg.fbs (hunk 1 above)
[the remark this answers is not in the document]
Comment (hayd) [reply src/msg.fbs]: @ry if we update to the Pinned[Buf] does that alleviate the zero copy concern?
(The streaming issue is a slightly larger problem, I'm not sure how to resolve it... can rust access ReadableStreams? what streamable type are you thinking of here?)

**Comment by nchavez324** [comment]
2 questions from a prospective implementer:

- Does the streamable resource type exist, or does that still have to be created? If not, where's the best place to start reading and hacking?
- By:
> if using WASM for this wouldn't be a better means
do you mean implementing the crypto algos in Rust outside of Deno and then invoking them in userland?

**Pull request closed without merging**